HDFOutputer.cc

```diff
@@ -8,6 +8,13 @@
 #include <cstring>
 #include <cmath>
 #include <set>
+#include <hdf5_hl.h>
+#include "H5Timing.h"
+#include "multidataset_plugin.h"
+
+int max_batch_size = 2;
+int hdf_method = -1;
+int total_n_events = -1;
```

Comment:
Please do not use global variables to control such things. Globals are detremental to large scale/component based programming.

Reply:
OK, we can move these to class fields